manifest.ts

```diff
@@ -0,0 +1,29 @@
+import {AppsBindings} from 'mattermost-redux/constants/apps';
+
+type RequestedPermissions = 'act_as_user' | 'act_as_bot'
+type RequestedLocations = AppsBindings.POST_MENU_ITEM | AppsBindings.CHANNEL_HEADER_ICON | AppsBindings.COMMAND | AppsBindings.IN_POST
+
+export type Manifest = {
+    app_id: string;
+    display_name: string;
+    description: string;
+    root_url: string;
+    requested_permissions: RequestedPermissions[];
+    oauth2_callback_url: string;
+    homepage_url: string;
+    requested_locations: RequestedLocations[];
+}
+
+export function getManifest(): Manifest {
```

Comment:
This should be converted to a manifest class which accepts a manifest type.  Constructor reads and verifies fields as required.  Simple helper methods to return fields.

Where would this exist so it could be imported?

Reply:
https://github.com/mattermost/mattermost-app-zendesk/issues/11#issuecomment-751797277